app/renderer/js/utils/domain-util.js

```diff
@@ -122,69 +160,59 @@ class DomainUtil {
 			ignoreCerts
 		};
 
-		return new Promise((resolve, reject) => {
-			request(checkDomain, (error, response) => {
+		return new Promise(async (resolve, reject) => {
```

Comment:
Just make the `checkDomain` function async. You will to throw error where there is `Promise.reject()`.